core/src/main/java/org/testcontainers/images/builder/ImageFromDockerfile.java

```diff
@@ -167,15 +168,31 @@ private void prePullDependencyImages(Set<String> imagesToPull) {
         final DockerClient dockerClient = DockerClientFactory.instance().client();
 
         imagesToPull.forEach(imageName -> {
+            String resolvedImageName = resolveImageNameFromBuildArgs(imageName);
             try {
-                log.info("Pre-emptively checking local images for '{}', referenced via a Dockerfile. If not available, it will be pulled.", imageName);
-                DockerClientFactory.instance().checkAndPullImage(dockerClient, imageName);
+                log.info("Pre-emptively checking local images for '{}', referenced via a Dockerfile. If not available, it will be pulled.", resolvedImageName);
+                DockerClientFactory.instance().checkAndPullImage(dockerClient, resolvedImageName);
             } catch (Exception e) {
-                log.warn("Unable to pre-fetch an image ({}) depended upon by Dockerfile - image build will continue but may fail. Exception message was: {}", imageName, e.getMessage());
+                log.warn("Unable to pre-fetch an image ({}) depended upon by Dockerfile - image build will continue but may fail. Exception message was: {}", resolvedImageName, e.getMessage());
             }
         });
     }
 
+    private String resolveImageNameFromBuildArgs(final String imageName) {
+        Optional<String> imageBuildArg = buildArgs.keySet()
+            .stream()
+            .filter(arg -> imageName.contains(arg))
+            .findFirst();
```

Comment:
This will only resolve a single match per FROM statement, right?
It also doesn't resolve the simple `$VARIABLE` pattern.

Reply:
Yep good spot, it would need to check more than one - could well have multiple args for a image depending on how fine grained your need is

Reply:
Also just seen your PR - I had been busy working on other projects and had forgot about this! I saw the notifications a few days ago so was going to see what was needed.
But happy for this to be closed in favour of your PR :)